### PR TITLE
Reduce size of GPBDictionary by getting rid of class creation methods

### PR DESCRIPTION
**Pull request by dmaclach**

### CODE REVIEW

File: objectivec/Tests/GPBMessageTests.m [hunk 1]
```diff
@@ -1238,7 +1238,8 @@ - (void)testAutocreatedMapCopy {
   // with different objects that are equal).
   TestRecursiveMessageWithRepeatedField *message3 =
       [TestRecursiveMessageWithRepeatedField message];
-  message3.iToI = [GPBInt32Int32Dictionary dictionaryWithInt32:10 forKey:20];
+  message3.iToI = [[GPBInt32Int32Dictionary alloc] init];
```
Comment (thomasvl) [inline objectivec/Tests/GPBMessageTests.m]: autorelease

File: objectivec/Tests/GPBMessageTests.m [hunk 2]
```diff
@@ -1323,7 +1324,8 @@ - (void)testReplaceAutocreatedMap {
     XCTAssertFalse([message hasA]);
     GPBInt32Int32Dictionary *iToI = [message.a.iToI retain];
     XCTAssertEqual(iToI->_autocreator, message.a);  // Pointer comparision
-    message.a.iToI = [GPBInt32Int32Dictionary dictionaryWithInt32:6 forKey:7];
+    message.a.iToI = [[GPBInt32Int32Dictionary alloc] init];
```
Comment (thomasvl) [inline objectivec/Tests/GPBMessageTests.m]: autorelease

File: objectivec/Tests/GPBMessageTests.m (hunk 1 above)
Comment (dmaclach) [reply objectivec/Tests/GPBMessageTests.m]: done

File: objectivec/Tests/GPBMessageTests.m (hunk 2 above)
Comment (dmaclach) [reply objectivec/Tests/GPBMessageTests.m]: done

**Pull request merged**